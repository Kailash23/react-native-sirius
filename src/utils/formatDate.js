const moment = require('moment');

moment.updateLocale('en', {
  relativeTime: {
    h: '1 hour',
    d: '1 day'
  }
});

export const formatDateForAggregation = epochDate => {
  const date = new Date(epochDate);

  const now = moment();
  const moDate = moment.utc(date);

  const diff = now.diff(moDate, 'days');

  if (diff < 7) {
    return moDate.fromNow();
  } else {
    return moDate.format('MMMM D, YYYY');
  }
};

export const formatDateforArticle = epochDate => {
  const date = new Date(epochDate);

  const moDate = moment.utc(date);

  return moDate.format('dddd, MMMM D, YYYY');
};

export const isOverOneYearOld = epochDate => {
  const date = new Date(epochDate);
  const today = new Date();

  return date.getYear() > today.getYear();
};

export const formatDateforArticleAnalytics = epochDate => {
  const date = new Date(epochDate);

  const moDate = moment.utc(date);

  return moDate.format('ddd MMMM D YYYY');
};
