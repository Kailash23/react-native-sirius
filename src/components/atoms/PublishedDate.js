import React from 'react';
import { withTheme, Paragraph } from 'react-native-paper';
import { customFonts } from '@styles/customFonts';
import { SIZES, HalfCardImageWidth, getFontSize } from '@styles/sizes';
import { fromUnixTime, format } from 'date-fns';
import { formatDateForAggregation } from '@utils/formatDate';
import { localStrings } from '@utils/i18n';

function PublishedDate(props) {
  const {
    lastmodifiedepoch,
    publishedepoch,
    half,
    paddingBottom,
    paddingTop,
    style,
    publishedDate
  } = props;
  let date = '';
  if (publishedepoch != undefined && publishedepoch > 0) {
    date = formatDateForAggregation(publishedepoch);
  } else if (lastmodifiedepoch != undefined && lastmodifiedepoch > 0) {
    date = formatDateForAggregation(lastmodifiedepoch);
  } else if (publishedDate) {
    const dateStr = publishedDate.replace(/ /g, ',');
    const shDate = format(new Date(dateStr), localStrings('shortDate'));
    date = shDate != 'Invalid Date' ? shDate : null;
  }

  if (date) {
    return (
      <Paragraph
        style={[
          {
            fontFamily: customFonts.merriweatherSansLight,
            fontSize: getFontSize(SIZES.FONT - 1),
            width: half ? HalfCardImageWidth - 32 : '100%',
            paddingTop: paddingTop ? paddingTop : 0,
            paddingBottom: paddingBottom ? paddingBottom : 0
          },
          { ...style }
        ]}>
        {date.toUpperCase()}
      </Paragraph>
    );
  } else {
    return null;
  }
}

export default PublishedDate;
