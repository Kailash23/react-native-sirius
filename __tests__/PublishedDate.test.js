import PublishedDate from '@atoms/PublishedDate';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
const moment = require('moment');
import React from 'react';
import { formatDateForAggregation } from '@utils/formatDate';

configure({ adapter: new Adapter() });

describe('Testing <PublishedDate/>', () => {
  test('should render a PublishedDate with proper dates', () => {
    const wrapper = shallow(
      <PublishedDate
        lastmodifiedepoch={1592867400000}
        publishedepoch={1592867400000}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  describe('Determine when the article was published.', () => {
    test('formatDateforArticle(*) should return when the article was published (1 day ago )', () => {
      expect(formatDateForAggregation(moment().subtract(1, 'day'))).toBe(
        '1 day ago'
      );
    });
  });
  describe('Determine when the article was published.', () => {
    test('formatDateforArticle(*) should return when the article was published (5 days ago)', () => {
      expect(formatDateForAggregation(moment().subtract(5, 'day'))).toBe(
        '5 days ago'
      );
    });
  });
});
