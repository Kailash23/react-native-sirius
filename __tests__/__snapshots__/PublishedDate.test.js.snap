// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PublishedDate/> should render a PublishedDate with proper dates 1`] = `
<Paragraph
  style={
    Array [
      Object {
        "fontFamily": "MerriweatherSans-Light",
        "fontSize": 11,
        "paddingBottom": 0,
        "paddingTop": 0,
        "width": "100%",
      },
      Object {},
    ]
  }
>
  4 DAYS AGO
</Paragraph>
`;
